### PR TITLE
Update for Markdown 3.2, drop 3.0 support

### PR DESCRIPTION
**Pull request by cwdavies**
Fix ValueError: No item named "strong_em" exists raised in wagtail-regulations

## Removals

Removed the following from `regdown/__init__.py`
- STRONG_EM_RE
- emphasis2
- strong_em
- strong2
- strong

## Changes

- `em_strong` replaced by `em_strong2`

## Testing

1. tox

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by Scotchester** [comment]
Letting Markdown install >= 3.2 was causing an error in our wagtail-regulations tests related to this change in Python-Markdown 3.2: https://python-markdown.github.io/change_log/release-3.2/#em-and-strong-inline-processor-changes

We'll leave this open for @willbarton to approve, or alternately say "Hey, we could just at easily fix this in a way that lets us not pin to an older version of markdown."

**Comment by willbarton** [comment]
> "Hey, we could just at easily fix this in a way that lets us not pin to an older version of markdown."

I think I'd prefer that. I'll take a look.

**Comment by cwdavies** [comment]
@willbarton and @Scotchester I fixed this in a way that lets us not pin to an older version of markdown.

**Review by willbarton — CHANGES_REQUESTED** [review]
We need to change the [requirement for Markdown to be `>=3.2` in `setup.py`](https://github.com/cfpb/regdown/blob/master/setup.py#L9) and either change or remove [the Markdown 3.0 and 3.1 version factors in `tox.ini`](https://github.com/cfpb/regdown/blob/master/tox.ini#L19-L21). The tests won't run against Markdown 3.2 otherwise, and this breaks regdown's compatibility with 3.0 and 3.1.

With some quick edits to tox.ini to run the tests, it also looks like we have some failures with emphasis and pseudo forms. 

One last thing, while we're in `tox.ini`, we could remove Python 2.7 and 3.7 and add Python 3.8, so our matrix ends up with Markdown 3.2 against Python 3.6 and 3.8.

**Comment by willbarton** [comment]
For context, the reason that Regdown re-defines the strong/em matching in Python-Markdown is to remove support for underscores specifying strong/em. That's because we use them for pseudo forms (`__Foo` or `Foo: ___`) the way those forms are specified in the eCFR plain text (so that reg text can be copied and pasted in).

That should probably be added as a comment, I'm sorry it's not already.

File: regdown/__init__.py [hunk 1]
```diff
@@ -73,24 +67,13 @@ def extendMarkdown(self, md):
 
         # Replace all inlinePatterns that include an underscore with patterns
         # that do not include underscores.
-        md.inlinePatterns.deregister('strong_em')
         md.inlinePatterns.deregister('em_strong')
```
Comment (willbarton) [inline regdown/__init__.py]: It looks like Python-Markdown has [broken out underscores in their own pattern processor](https://github.com/Python-Markdown/markdown/blob/master/markdown/inlinepatterns.py#L94). In that case, I think we can just change this to:

```suggestion
        md.inlinePatterns.deregister('em_strong2')
```

And then delete lines [72-76](https://github.com/cfpb/regdown/pull/2/files#diff-75710316ba4d56dc8783f7639b8e5af7R72-R76).

File: regdown/__init__.py [hunk 2]
```diff
@@ -73,24 +67,13 @@ def extendMarkdown(self, md):
...
-        md.inlinePatterns.deregister('strong')
-        md.inlinePatterns.deregister('strong2')
-        md.inlinePatterns.deregister('emphasis2')
 
         md.inlinePatterns.register(
             DoubleTagInlineProcessor(EM_STRONG_RE, 'strong,em'),
             'em_strong',
             60
         )
```
Comment (willbarton) [inline regdown/__init__.py:72]: This whole pattern registration can be deleted.

**Comment by cwdavies** [comment]
Test Results from running tox:
```
...
regdown/__init__.py:9:1: F401 'markdown.inlinepatterns.SimpleTagInlineProcessor' imported but unused
ERROR: InvocationError for command /Users/DaviesC/Projects/CFPB/regdown/.tox/lint/bin/flake8 regdown (exited with code 1)
...
test_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... /Users/DaviesC/Projects/CFPB/regdown/.tox/py36-md32/lib/python3.6/site-packages/markdown/blockparser.py:123: DeprecationWarning: 'etree' is deprecated. Use 'xml.etree.ElementTree' instead.
  if processor.run(parent, blocks) is not False:
ok
...
test_no_underscore_emphasis (regdown.tests.RegulationsExtensionTestCase) ... FAIL
...
test_pseudo_form_field_end_of_line (regdown.tests.RegulationsExtensionTestCase) ... /Users/DaviesC/Projects/CFPB/regdown/.tox/py36-md32/lib/python3.6/site-packages/markdown/treeprocessors.py:285: DeprecationWarning: 'etree' is deprecated. Use 'xml.etree.ElementTree' instead.
  node = pattern.handleMatch(match)
ok
...
FAIL: test_no_underscore_emphasis (regdown.tests.RegulationsExtensionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/DaviesC/Projects/CFPB/regdown/regdown/tests.py", line 258, in test_no_underscore_emphasis
    self.assertIn('<em>foo</em>', regdown('*foo*'))
AssertionError: '<em>foo</em>' not found in '<p class="regdown-block" data-label="" id="f8c728521aa1e64b9f12891110d4c6982f8cd4786540df03f75b6a7e">*foo*</p>'

Ran 30 tests in 0.057s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=30 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=30 errors=0 failures=1>
ERROR: InvocationError for command /Users/DaviesC/Projects/CFPB/regdown/.tox/py36-md32/bin/coverage run --source=regdown setup.py test (exited with code 1)

```

**Comment by cwdavies** [comment]
Test Results from running tox:
```
...
py36-md32 run-test: commands[0] | coverage erase
py36-md32 run-test: commands[1] | coverage run --source=regdown setup.py test
running test
running egg_info
writing regdown.egg-info/PKG-INFO
writing dependency_links to regdown.egg-info/dependency_links.txt
writing requirements to regdown.egg-info/requires.txt
writing top-level names to regdown.egg-info/top_level.txt
reading manifest file 'regdown.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'regdown.egg-info/SOURCES.txt'
running build_ext
test_extract_labeled_paragraph (regdown.tests.RegdownUtilsTestCase) ... ok
test_extract_labeled_paragraph_not_found (regdown.tests.RegdownUtilsTestCase) ... ok
test_extract_labeled_paragraph_startswith (regdown.tests.RegdownUtilsTestCase) ... ok
test_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... /Users/DaviesC/Projects/CFPB/regdown/.tox/py36-md32/lib/python3.6/site-packages/markdown/blockparser.py:123: DeprecationWarning: 'etree' is deprecated. Use 'xml.etree.ElementTree' instead.
  if processor.run(parent, blocks) is not False:
ok
test_block_reference (regdown.tests.RegulationsExtensionTestCase) ... ok
test_block_reference_no_contents (regdown.tests.RegulationsExtensionTestCase) ... ok
test_block_reference_renderer_not_callable (regdown.tests.RegulationsExtensionTestCase) ... ok
test_block_reference_resolver_not_callable (regdown.tests.RegulationsExtensionTestCase) ... ok
test_complex_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_complicated_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_deeper_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_emdash (regdown.tests.RegulationsExtensionTestCase) ... ok
test_empty_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_even_deeper_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_linebreak_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_list_state (regdown.tests.RegulationsExtensionTestCase) ... ok
test_makeExtension (regdown.tests.RegulationsExtensionTestCase)
Test that Markdown can load our extension from a string ... ok
test_multi_part_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_multiple_linebreaks_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_no_underscore_emphasis (regdown.tests.RegulationsExtensionTestCase) ... /Users/DaviesC/Projects/CFPB/regdown/.tox/py36-md32/lib/python3.6/site-packages/markdown/treeprocessors.py:285: DeprecationWarning: 'etree' is deprecated. Use 'xml.etree.ElementTree' instead.
  node = pattern.handleMatch(match)
ok
test_nolabel (regdown.tests.RegulationsExtensionTestCase) ... ok
test_prefixed_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_end_of_line (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_inside_line (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_number_of_underscores (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_start_of_line (regdown.tests.RegulationsExtensionTestCase) ... ok
test_section_symbol_with_non_breaking_space (regdown.tests.RegulationsExtensionTestCase) ... ok
test_tables_extension_exists (regdown.tests.RegulationsExtensionTestCase) ... ok

----------------------------------------------------------------------
Ran 30 tests in 0.057s

OK
py36-md32 run-test: commands[2] | coverage report -m
Name                  Stmts   Miss  Cover   Missing
---------------------------------------------------
regdown/__init__.py     113      0   100%
regdown/tests.py        128      0   100%
---------------------------------------------------
TOTAL                   241      0   100%
...
py38-md32 run-test: commands[0] | coverage erase
py38-md32 run-test: commands[1] | coverage run --source=regdown setup.py test
running test
running egg_info
writing regdown.egg-info/PKG-INFO
writing dependency_links to regdown.egg-info/dependency_links.txt
writing requirements to regdown.egg-info/requires.txt
writing top-level names to regdown.egg-info/top_level.txt
reading manifest file 'regdown.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'regdown.egg-info/SOURCES.txt'
running build_ext
test_extract_labeled_paragraph (regdown.tests.RegdownUtilsTestCase) ... ok
test_extract_labeled_paragraph_not_found (regdown.tests.RegdownUtilsTestCase) ... ok
test_extract_labeled_paragraph_startswith (regdown.tests.RegdownUtilsTestCase) ... ok
test_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... /Users/DaviesC/Projects/CFPB/regdown/.tox/py38-md32/lib/python3.8/site-packages/markdown/blockparser.py:123: DeprecationWarning: 'etree' is deprecated. Use 'xml.etree.ElementTree' instead.
  if processor.run(parent, blocks) is not False:
ok
test_block_reference (regdown.tests.RegulationsExtensionTestCase) ... ok
test_block_reference_no_contents (regdown.tests.RegulationsExtensionTestCase) ... ok
test_block_reference_renderer_not_callable (regdown.tests.RegulationsExtensionTestCase) ... ok
test_block_reference_resolver_not_callable (regdown.tests.RegulationsExtensionTestCase) ... ok
test_complex_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_complicated_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_deeper_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_emdash (regdown.tests.RegulationsExtensionTestCase) ... ok
test_empty_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_even_deeper_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_linebreak_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_list_state (regdown.tests.RegulationsExtensionTestCase) ... ok
test_makeExtension (regdown.tests.RegulationsExtensionTestCase)
Test that Markdown can load our extension from a string ... ok
test_multi_part_appendix_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_multiple_linebreaks_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_no_underscore_emphasis (regdown.tests.RegulationsExtensionTestCase) ... /Users/DaviesC/Projects/CFPB/regdown/.tox/py38-md32/lib/python3.8/site-packages/markdown/treeprocessors.py:285: DeprecationWarning: 'etree' is deprecated. Use 'xml.etree.ElementTree' instead.
  node = pattern.handleMatch(match)
ok
test_nolabel (regdown.tests.RegulationsExtensionTestCase) ... ok
test_prefixed_inline_interp_label (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_end_of_line (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_inside_line (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_number_of_underscores (regdown.tests.RegulationsExtensionTestCase) ... ok
test_pseudo_form_field_start_of_line (regdown.tests.RegulationsExtensionTestCase) ... ok
test_section_symbol_with_non_breaking_space (regdown.tests.RegulationsExtensionTestCase) ... ok
test_tables_extension_exists (regdown.tests.RegulationsExtensionTestCase) ... ok

----------------------------------------------------------------------
Ran 30 tests in 0.044s

OK
py38-md32 run-test: commands[2] | coverage report -m
Name                  Stmts   Miss  Cover   Missing
---------------------------------------------------
regdown/__init__.py     113      0   100%
regdown/tests.py        128      0   100%
---------------------------------------------------
TOTAL                   241      0   100%

  lint: commands succeeded
  py36-md32: commands succeeded
  py38-md32: commands succeeded
  congratulations :)
```

**Review by willbarton — APPROVED** [review]
Looks great to me!

**Pull request merged**